### PR TITLE
Restructure `frame_benchmarking` macro related exports

### PR DESCRIPTION
**Pull request by juangirini**
Partial for https://github.com/paritytech/polkadot-sdk/issues/172

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412356

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412358

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412417

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412415

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412416

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412526

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3412525

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3413503

**Comment by juangirini** [comment]
bot rebaes

**Comment by juangirini** [comment]
bot rebase

File: frame/timestamp/Cargo.toml [hunk 1]
```diff
@@ -44,6 +45,7 @@ std = [
 	"sp-io?/std",
 	"sp-runtime/std",
 	"sp-std/std",
+  "sp-storage/std",
```
Comment (ggwpez) [inline frame/timestamp/Cargo.toml:48]:
```suggestion
	 "sp-storage/std",
```
CI noticed :sunglasses:

**Review by ggwpez — APPROVED** [review]

File: frame/benchmarking/src/lib.rs [hunk 2]
```diff
@@ -28,29 +28,27 @@ mod tests_instance;
...
+	pub use paste;
+	pub use sp_core::defer;
+	pub use sp_io::storage::root as storage_root;
+	pub use sp_runtime::{traits::Zero, StateVersion};
+	pub use sp_std::{self, boxed::Box, str, vec, vec::Vec};
+	pub use sp_storage::{well_known_keys, TrackedStorageKey};
+}
 
 #[cfg(feature = "std")]
 pub use analysis::{Analysis, AnalysisChoice, BenchmarkSelector};
```
[the remark this answers is not in the document]
Comment (juangirini) [reply frame/benchmarking/src/lib.rs:50]: It seems to be its own hairy little monster. I have created a separate issue for it https://github.com/paritytech/substrate/issues/14822.

File: frame/support/test/tests/construct_runtime_ui/number_of_pallets_exceeds_tuple_size.stderr [hunk 3]
```diff
@@ -55,7 +55,9 @@ help: you might have meant to use the associated type
    |
 38 |     type PalletInfo = Self::PalletInfo;
    |                       ~~~~~~~~~~~~~~~~
-help: consider importing this trait
+help: consider importing one of these items
+   |
+1  + use frame_benchmarking::__private::traits::PalletInfo;
```
Comment (liamaharon) [inline frame/support/test/tests/construct_runtime_ui/number_of_pallets_exceeds_tuple_size.stderr:60]: Shouldn't this suggest

```suggestion
help: consider importing one of these items
   |
1  + use frame_support::traits::PalletInfo;
```

**Review by liamaharon — COMMENTED** [review]
Looking good, just not sure if a stderr log needs to be adjusted.

File: frame/support/test/tests/construct_runtime_ui/number_of_pallets_exceeds_tuple_size.stderr (hunk 3 above)
Comment (juangirini) [reply frame/support/test/tests/construct_runtime_ui/number_of_pallets_exceeds_tuple_size.stderr:60]: Interesting, do you know if it is possible to somehow tell the compiler to ignore that option?

File: bin/node-template/pallets/template/src/benchmarking.rs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 //! Benchmarking setup for pallet-template
 #![cfg(feature = "runtime-benchmarks")]
 use super::*;
+use sp_std::vec;
```
Comment (bkchr) [inline bin/node-template/pallets/template/src/benchmarking.rs:4]: This looks wrong? I don't see any usage of the macro in here. Maybe some macro needs this? If yes, it should pull this in automatically.

File: frame/benchmarking/src/baseline.rs [hunk 5]
```diff
@@ -20,13 +20,13 @@
...
 #![cfg(feature = "runtime-benchmarks")]
 
-use super::*;
 use crate::benchmarks;
 use frame_system::Pallet as System;
 use sp_runtime::{
 	traits::{AppVerify, Hash},
 	RuntimeAppPublic,
 };
+use sp_std::{vec, vec::Vec};
```
Comment (bkchr) [inline frame/benchmarking/src/baseline.rs:29]: Here again?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434488

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434487

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434489

File: frame/benchmarking/src/baseline.rs (hunk 5 above)
Comment (juangirini) [reply frame/benchmarking/src/baseline.rs]: It is actually required https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434492
```
[2023-08-24 12:02:10]   error: cannot find macro `vec` in this scope
[2023-08-24 12:02:10]     --> /builds/parity/mirrors/substrate/frame/benchmarking/src/baseline.rs:93:14
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   93 |             let msg = vec![j, j];
[2023-08-24 12:02:10]      |                       ^^^
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]      = help: consider importing one of these items:
[2023-08-24 12:02:10]              crate::__private::vec
[2023-08-24 12:02:10]              scale_info::prelude::vec
[2023-08-24 12:02:10]              sp_api::vec
[2023-08-24 12:02:10]              sp_std::vec
[2023-08-24 12:02:10] 
[2023-08-24 12:02:10]   error[E0412]: cannot find type `Vec` in this scope
[2023-08-24 12:02:10]     --> /builds/parity/mirrors/substrate/frame/benchmarking/src/baseline.rs:92:21
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   92 |         let msg_and_sigs: Vec<_> = (0..sigs_count).map(|j| {
[2023-08-24 12:02:10]      |                           ^^^ not found in this scope
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   help: consider importing one of these items
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   23 + use crate::__private::Vec;
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   23 + use frame_support::dispatch::Vec;
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   23 + use scale_info::prelude::vec::Vec;
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]   23 + use sp_api::vec::Vec;
[2023-08-24 12:02:10]      |
[2023-08-24 12:02:10]        and 2 other candidates
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434673

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434674

File: bin/node-template/pallets/template/src/benchmarking.rs (hunk 4 above)
Comment (juangirini) [reply bin/node-template/pallets/template/src/benchmarking.rs]: It is expected here https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3434676
```
[2023-08-24 12:21:48]   error: cannot find macro `vec` in this scope
[2023-08-24 12:21:48]     --> /builds/parity/mirrors/substrate/bin/node-template/pallets/template/src/benchmarking.rs:10:1
[2023-08-24 12:21:48]      |
[2023-08-24 12:21:48]   10 | #[benchmarks]
[2023-08-24 12:21:48]      | ^^^^^^^^^^^^^
[2023-08-24 12:21:48]      |
[2023-08-24 12:21:48]      = help: consider importing one of these items:
[2023-08-24 12:21:48]              crate::benchmarking::__private::vec
[2023-08-24 12:21:48]              frame_benchmarking::__private::vec
[2023-08-24 12:21:48]              scale_info::prelude::vec
[2023-08-24 12:21:48]              sp_std::vec
[2023-08-24 12:21:48]      = note: this error originates in the attribute macro `benchmarks` (in Nightly builds, run with -Z macro-backtrace for more info)
```

File: frame/support/test/tests/construct_runtime_ui/number_of_pallets_exceeds_tuple_size.stderr (hunk 3 above)
Comment (liamaharon) [reply frame/support/test/tests/construct_runtime_ui/number_of_pallets_exceeds_tuple_size.stderr:60]: maybe @sam0x17 knows

**Review by liamaharon — APPROVED** [review]

**Comment by juangirini** [comment]
bot merge

**Pull request merged**